ibis/pandas/execution/window.py

```diff
@@ -161,14 +161,17 @@ def trim_with_timecontext(data, timecontext: Optional[TimeContext]):
     if not timecontext:
         return data
 
-    df = data.reset_index()
+    df = data.reset_index(level=1)
```

Comment:
Why this change?

Reply:
OK this is tricky. Here, data is a Series of MultiIndex (int, Timestamp) where int is some index (0,1,2,3 ...) and Timestamp column is our time column. When we filter by time, the index is also filtered so we will have something like:
```
index  time  value
2       2010-01-02 2
3       2010-01-03 3
```
As you can see the index may not start from 0.  And later if we put this Series back to generate result. There will be NaN and NaT filling in, since pandas will align rows according to index number.
Therefore, all changes in this function is trying to reindex the int inside the Multiindex after filtering, to guarantee the correctness of our result.

Reply:
Ok, I think this needs to be explained in the code, otherwise it's kind of confusing.

Also, is it always the case that time is an index here? What if this is not a moving window, e.g.

```
win = window(group_by='id')
```

What's the behavior of this code in that case?

Reply:
Or perhaps

```
win = window(preceding=interval(days=1), group_by='id', order_by='some_other_time_column')
```

?

Reply:
Or

```
win = window(preceding=10, group_by='id', order_by='time')
```

Reply:
In general, I think this function is not very clear about what the expected input and output. It would be helpful to clarify.